prow/pod-utils/clone/clone.go

```diff
@@ -19,18 +19,28 @@ package clone
 import (
 	"bytes"
 	"fmt"
+	"net/url"
 	"os/exec"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
 	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/logrusutil"
 )
 
 // Run clones the refs under the prescribed directory and optionally
 // configures the git username and email in the repository as well.
-func Run(refs prowapi.Refs, dir, gitUserName, gitUserEmail, cookiePath string, env []string) Record {
+func Run(refs prowapi.Refs, dir, gitUserName, gitUserEmail, cookiePath string, env []string, oauthToken string) Record {
+	if len(oauthToken) > 0 {
+		logrus.SetFormatter(logrusutil.NewCensoringFormatter(logrus.StandardLogger().Formatter, func() sets.String {
```

Comment:
What if the current formatter is not the standard formatter, but the JSON one?

Reply:
You are right. It seems that we are already using the JSON formatter as the default one, so there is no reason to change this here.

Reply:
Oh it seems that this will still use the JSONFormatter since it was initialized [here](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/clonerefs/main.go#L28) and [here](https://github.com/kubernetes/test-infra/blob/master/prow/logrusutil/logrusutil.go#L46) we declare it as a default.

Reply:
Won't the second call to `SetFormatter` overwrite the previous one?

Reply:
No. I tested it also to confirm it.